aten/src/ATen/native/native_functions.yaml

```diff
@@ -818,10 +818,22 @@
   variants: function, method
 
 - func: fill_(Tensor(a!) self, Scalar value) -> Tensor(a!)
+  cpu_half: True
+  cpu_bool: True
+  cuda_bool: True
   variants: function, method
+  dispatch:
+    CPU: fill__cpu
+    CUDA: fill__cuda
 
 - func: fill_(Tensor(a!) self, Tensor value) -> Tensor(a!)
+  cpu_half: True
+  cpu_bool: True
+  cuda_bool: True
   variants: function, method
+  dispatch:
+    CPU: fill__cpu
```

Comment:
I know this was mentioned before, but is there a reason we can't do the dispatch directly here?